src/server/generic_family.cc

```diff
@@ -515,7 +516,7 @@ OpResult<bool> OnRestore(const OpArgs& op_args, std::string_view key, std::strin
 bool ScanCb(const OpArgs& op_args, PrimeIterator it, const ScanOpts& opts, StringVec* res) {
   auto& db_slice = op_args.shard->db_slice();
   if (it->second.HasExpire()) {
-    it = db_slice.ExpireIfNeeded(op_args.db_cntx, it).first;
```

Comment:
I think we are missing a preupdate call on the scan items. If we expire the entry while scanning than how do we make sure we call PreUpdate before we delete it?

Reply:
I believe we have the same problem with Eviction and Expiring items on background process

Reply:
lets open another github issue on this

Reply:
I remember writing a reply but I don't see it now.
I think this should be handled in ExpireIfNeeded(). I'll file a bug.

Reply:
https://github.com/dragonflydb/dragonfly/issues/2316